core/domain/exp_domain.py

```diff
@@ -4163,7 +4145,19 @@ def create_mapping_for_new_version(
         # Go through each state id and name and compare the new state
         # with its corresponding old state to find whether significant
         # change has happened or not.
+        state_ids_to_be_removed = []
         for (state_id, state_name) in state_ids_to_names.iteritems():
+            if state_name == 'END' and state_name not in new_exploration.states:
+                # If previous version of exploration has END state but new
+                # version does not have END state. In addition, there is no
+                # commit command corresponding to rename or removal of END
+                # state then END state was added to previouos exploratio
+                # verison by states dict schema migration from v2 to v3 and it
+                # was not added to new exploration version by the migration as
+                # new version does not have any rules having 'END' state as
+                # their destination state name.
```

Comment:
Here and above: this documentation may be a bit hard to understand/read. Can you split it into multiple sentences to make it a bit clearer?

Reply:
Added more clearer explanation.